shell/src/main/java/alluxio/cli/ValidateHdfsMount.java

```diff
@@ -0,0 +1,233 @@
+import org.apache.commons.lang.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * A tool to validate an HDFS mount, before the paths is mounted to Alluxio.
+ * */
+public class ValidateHdfsMount {
+  private static final Logger LOG = LoggerFactory.getLogger(ValidateHdfsMount.class);
+  private static final String JSON_START_POS_MARKER = "ValidateHdfsMount task results: \n";
+
+  private static final Option READONLY_OPTION =
+          Option.builder()
+                  .longOpt("readonly")
+                  .required(false)
+                  .hasArg(false)
+                  .desc("mount point is readonly in Alluxio")
+                  .build();
+  private static final Option SHARED_OPTION =
+          Option.builder()
+                  .longOpt("shared")
+                  .required(false)
+                  .hasArg(false)
+                  .desc("mount point is shared")
+                  .build();
+  private static final Option OPTION_OPTION =
+          Option.builder()
+                  .longOpt("option")
+                  .required(false)
+                  .hasArg(true)
+                  .numberOfArgs(2)
+                  .argName("key=value")
+                  .valueSeparator('=')
+                  .desc("options associated with this mount point")
+                  .build();
+  private static final Option LOCAL_OPTION =
+          Option.builder().required(false).longOpt("local").hasArg(false)
+                  .desc("running only on localhost").build();
+  private static final Options OPTIONS =
+          new Options().addOption(READONLY_OPTION).addOption(SHARED_OPTION)
+                  .addOption(OPTION_OPTION).addOption(LOCAL_OPTION);
+
+  /**
+   * Invokes {@link UnderFileSystemContractTest} to validate UFS operations.
+   *
+   * @param path the UFS path
+   * @param conf the UFS conf
+   * @return a {@link alluxio.cli.ValidateUtils.TaskResult} containing the validation result
+   *        of the UFS operations
+   * */
+  public static ValidateUtils.TaskResult runUfsTests(String path, InstancedConfiguration conf) {
+    try {
+      UnderFileSystemContractTest test = new UnderFileSystemContractTest(path, conf);
+      return test.runValidationTask();
+    } catch (IOException e) {
+      return new ValidateUtils.TaskResult(ValidateUtils.State.FAILED, "ufsTests",
+              ValidateUtils.getErrorInfo(e), "");
+    }
+  }
+
+  /**
+   * The entrance.
+   *
+   * @param args command line arguments
+   * */
+  public static void main(String[] args) throws Exception {
+    // TODO(jiacheng): use jccommand?
```

Comment:
After some tests I find JCommand to be better. Will refactor this to use JCommand.

Reply:
Not really, i'll remove this TODO. I found the existing cmdline works just fine. JCommand is not a better option anymore.